table/iterator.go

```diff
@@ -85,169 +107,52 @@ var (
 	current = 1
 )
 
-func (itr *blockIterator) getKey(idx int) []byte {
-	y.AssertTrue(idx >= 0 && idx < itr.numEntries)
-
-	idxPos := itr.entryOffsets[idx]
-	var h header
-	idxPos += uint32(h.Decode(itr.data[idxPos:]))
-
-	// Convert to int before adding to avoid uint16 overflow.
-	idxKey := make([]byte, int(h.plen)+int(h.klen))
-	copy(idxKey, itr.baseKey[:h.plen])
-	copy(idxKey[h.plen:], itr.data[idxPos:idxPos+uint32(h.klen)])
-
-	return idxKey
-}
-
-// Seek brings us to the first block element that is >= input key.
-func (itr *blockIterator) Seek(key []byte, whence int) {
+// seek brings us to the first block element that is >= input key.
+func (itr *blockIterator) seek(key []byte, whence int) {
 	itr.err = nil
 	startIndex := 0 // This tells from which index we should start binary search.
 
 	switch whence {
 	case origin:
-		itr.Reset()
+		// We don't need to do anything. startIndex is already at 0
 	case current:
-		startIndex = itr.currentIdx
+		startIndex = itr.idx
 	}
 
-	itr.Init() // If iterator is not initialized or has been reset.
-
-	idx := sort.Search(itr.numEntries, func(idx int) bool {
+	foundEntryIdx := sort.Search(len(itr.entryOffsets), func(idx int) bool {
```

Comment:
I would like to keep it as it is since the added complexity might not be worth it. Also, we never use the `startindex`. We added it a while ago but never actually used it.